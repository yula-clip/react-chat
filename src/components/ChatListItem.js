import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import moment from 'moment';
import { Link } from 'react-router-dom';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from './Avatar';

const styles = theme => ({
  activeItem: {
    backgroundColor: theme.palette.grey[200],
  }
});

const ChatListItem = ({ classes, title, disabled, chatId, active, createdAt }) => (
  <ListItem 
    button
    component={Link}
    to={`/chat/${chatId}`}
    className={active ? classes.activeItem : ''}
    disabled={disabled}
  >
    <Avatar colorFrom = {chatId}>{title}</Avatar>
    <ListItemText primary={title} secondary={moment(createdAt).fromNow()}/>
  </ListItem>
);

export default withStyles(styles)(ChatListItem);
