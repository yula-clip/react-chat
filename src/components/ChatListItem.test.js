/* eslint-env jest */
import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import ChatListItem from "./ChatListItem";

jest.mock("moment", () => () => ({ fromNow: () => "2 days ago" }));

const mockProps = {
  disabled: false,
  active: false,
  chatId: "12345",
  title: "My Chat",
  createdAt: "2018-03-16T10:53:23.200Z"
};

describe("<ChatList />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <ChatListItem {...mockProps} />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ChatListItem {...mockProps} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders disabled", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ChatListItem {...mockProps} disabled />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders active", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ChatListItem {...mockProps} active />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
