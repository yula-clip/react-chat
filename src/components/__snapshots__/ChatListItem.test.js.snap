// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatList /> renders active 1`] = `
<a
  className="MuiButtonBase-root-81 MuiListItem-root-69 MuiListItem-default-72 MuiListItem-gutters-77 MuiListItem-button-78 ChatListItem-activeItem-68"
  href="/chat/12345"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="MuiAvatar-root-84 MuiAvatar-colorDefault-85"
    style={
      Object {
        "backgroundColor": "#f44336",
      }
    }
  >
    MC
  </div>
  <div
    className="MuiListItemText-root-87"
  >
    <span
      className="MuiTypography-root-93 MuiTypography-subheading-100 MuiListItemText-primary-90"
    >
      My Chat
    </span>
    <p
      className="MuiTypography-root-93 MuiTypography-body1-102 MuiTypography-colorTextSecondary-126 MuiListItemText-secondary-91"
    >
      2 days ago
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-128"
  />
</a>
`;

exports[`<ChatList /> renders correctly 1`] = `
<a
  className="MuiButtonBase-root-81 MuiListItem-root-69 MuiListItem-default-72 MuiListItem-gutters-77 MuiListItem-button-78"
  href="/chat/12345"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="MuiAvatar-root-84 MuiAvatar-colorDefault-85"
    style={
      Object {
        "backgroundColor": "#f44336",
      }
    }
  >
    MC
  </div>
  <div
    className="MuiListItemText-root-87"
  >
    <span
      className="MuiTypography-root-93 MuiTypography-subheading-100 MuiListItemText-primary-90"
    >
      My Chat
    </span>
    <p
      className="MuiTypography-root-93 MuiTypography-body1-102 MuiTypography-colorTextSecondary-126 MuiListItemText-secondary-91"
    >
      2 days ago
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-128"
  />
</a>
`;

exports[`<ChatList /> renders disabled 1`] = `
<a
  className="MuiButtonBase-root-81 MuiButtonBase-disabled-82 MuiListItem-root-69 MuiListItem-default-72 MuiListItem-gutters-77 MuiListItem-disabled-75 MuiListItem-button-78"
  href="/chat/12345"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="-1"
>
  <div
    className="MuiAvatar-root-84 MuiAvatar-colorDefault-85"
    style={
      Object {
        "backgroundColor": "#f44336",
      }
    }
  >
    MC
  </div>
  <div
    className="MuiListItemText-root-87"
  >
    <span
      className="MuiTypography-root-93 MuiTypography-subheading-100 MuiListItemText-primary-90"
    >
      My Chat
    </span>
    <p
      className="MuiTypography-root-93 MuiTypography-body1-102 MuiTypography-colorTextSecondary-126 MuiListItemText-secondary-91"
    >
      2 days ago
    </p>
  </div>
</a>
`;
